# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [31]:
import pandas as pd
import numpy as np
from scipy import stats
import math

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

# Question 1.

## What test is appropriate here? Does the CLT apply?

A hypothesis testing on the difference of means for this data is applicable. A two sample z test can be used here since the sample size n is large and the t score and the z score will be approximately equal. In addition, if the sample size n is large, the standard deviation of the population can be approximated with the sample's standard deviation.

## Does the CLT apply?

The CLT does apply here since the normal condition is met as long as the sample size $n\geq 30$.

# Question 2.

## What are the null and alternative hypotheses?

The null and alternative hypotheses are stated as follows:

$H_{o}: \mu_{w} - \mu_{b} = 0$

$H_{a}: \mu_{w} - \mu_{b} \neq 0$

where the $\mu_{1}$ and $\mu_{2}$ are the black and white population true mean callbacks for jobs. To begin the hypothesis testing, the significance level needs to be preset. The significance level will be set to $\alpha = .05$. The dataset will be seperated into two groups, one white sounding names and the other black sounding names.

In [38]:
df_w = df[df.race == 'w']
df_b = df[df.race == 'b']
data.shape

2435.0

In [34]:
w_mean = np.mean(df_w.call)
w_std = np.std(df_w.call)
b_mean = np.mean(df_b.call)
b_std = np.std(df_b.call)
w_mean, w_std, b_mean, b_std

(0.09650924056768417,
 0.29528486728668213,
 0.0644763857126236,
 0.24559901654720306)

In [35]:
diff_mean = w_mean - b_mean
diff_mean

0.03203285485506058

In [27]:
n_w = len(df_w)
n_b = len(df_b)
n_w, n_b

(2435, 2435)

# Question 3.

## Compute margin of error, confidence interval, and p-value.

Below shows the formula used to calculate the z score where $s_{1}$ and $s_{2}$ are the samples' standard deviations from each group.

$S_{w} = \frac{s_{w}^2}{n_{w}}$

$S_{b} = \frac{s_{b}^2}{n_{b}}$

$Z score: \frac{\bar x_{w} - \bar x_{b} - \mu_{w} - \mu_{b}}{\sqrt{S_{w} + S_{b}}} $

Z score = 4.115583422082968

p value = 3.8620128019806188e-05

As seen below, the p value determined is 3.8620128019806188e-05, which means that the probability that difference in mean between the two samples here is so unlikely to have occured given the null hypothesis. Since the p value is less than the significance level, the null hypothesis is rejected and the alternative hypothesis is accepted.

Confidence Interval = [0.016777570469610682, 0.047288139240510473]

Margin of Error = 0.015255284385449893

In [48]:
s_w = w_std**2/n_w
s_b = b_std**2/n_b
z = diff_mean/math.sqrt(s_w + s_b)
p_value = 2*(1 - stats.norm.cdf(z))
z, p_value

(4.115583422082968, 3.8620128019806188e-05)

In [47]:
#This method came from the scipy library and was used to help backup my initial calculations
stats.ttest_ind(df_w.call,df_b.call)

Ttest_indResult(statistic=4.1147052908617514, pvalue=3.9408021031288859e-05)

In [57]:
# 95% Confidence Interval
diff_mean - 1.96*np.sqrt(s_w + s_b), diff_mean + 1.96*np.sqrt(s_w + s_b)

(0.016777570469610682, 0.047288139240510473)

In [54]:
margin_of_error = 1.96*np.sqrt(s_w + s_b)
margin_of_error

0.015255284385449893

# Question 4 and 5.

## Write a story describing the statistical significance in the context or the original problem. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

As a result, after performing the hypothesis testing, where the null hypothesis was that job seekers got callbacks regardless of whether they have black sounding names or white sounding names, the null hypothesis was rejected due to the fact that the p value computed was less than the significance level. This means that on average there is a 3.2%  disparity between job seekers who have black sounding name than those who have white sounding names. Although this may be the case, additional features will need to be included to determine if callbacks were primarily due to race. The dataset also contained education, years of experience, and many more. Without looking into other qualities, it is hard to definitive come to a conclusion.